File: examples/Java/wuserver.java
package guide;

import java.util.Random;

import org.zeromq.SocketType;
import org.zeromq.ZMQ;
import org.zeromq.ZContext;

//
//  Weather update server in Java
//  Binds PUB socket to tcp://*:5556
//  Publishes random weather updates
//
public class wuserver
{
    public static void main(String[] args) throws Exception
    {
        //  Prepare our context and publisher
        try (ZContext context = new ZContext()) {
            ZMQ.Socket publisher = context.createSocket(SocketType.PUB);
            publisher.bind("tcp://*:5556");
            publisher.bind("ipc://weather");

            //  Initialize random number generator
            Random srandom = new Random(System.currentTimeMillis());
            while (!Thread.currentThread().isInterrupted()) {
                //  Get values that will fool the boss
                int zipcode, temperature, relhumidity;
                zipcode = 10000 + srandom.nextInt(10000);
                temperature = srandom.nextInt(215) - 80 + 1;
                relhumidity = srandom.nextInt(50) + 10 + 1;

                //  Send message to all subscribers
                String update = String.format(
                    "%05d %d %d", zipcode, temperature, relhumidity
                );
                publisher.send(update, 0);
            }
        }
    }
}

File: examples/Clojure/wuserver.clj
;; Weather update server
;; Binds PUB socket to tcp://*:5556
;; Publishes random weather updates

(ns wuserver
  (:require [zeromq.zmq :as zmq]))

(defn -main []
  (let [context (zmq/zcontext)]
    (with-open [publisher (doto (zmq/socket context :pub)
                            (zmq/bind "tcp://*:5556")
                            (zmq/bind "ipc://weather.ipc"))]
      (while (not (.. Thread currentThread isInterrupted))
        (let [zipcode (rand-int 100000)
              temperature (- (rand-int 215) 80)
              relhumidity (+ (rand-int 50) 10)]
          (zmq/send-str publisher
                        (format "%05d %d %d" zipcode temperature relhumidity)))))))
